src/pagination.rs

```diff
@@ -212,10 +256,80 @@ fn sort_posts(posts: &mut Vec<&liquid::value::Value>, config: &PaginationConfig)
     })
 }
 
+fn distribute_posts_by_tags<'a>(
+    all_posts: &[&'a liquid::value::Value],
+) -> HashMap<String, Vec<&'a liquid::value::Value>> {
+    let mut per_tags: HashMap<String, Vec<&'a liquid::value::Value>> = HashMap::new();
+    let add_to_tag =
+        |tag: &str,
+         post: &'a liquid::value::Value,
+         per_tags: &mut HashMap<String, Vec<&'a liquid::value::Value>>| {
+            if per_tags.contains_key(tag) {
+                let v = per_tags
+                    .get_mut(tag)
+                    .expect("Tag's presence has been tested, can't happen!");
+                v.push(post)
+            } else {
+                per_tags.insert(tag.to_string(), vec![post]);
+            }
+        };
+    all_posts.into_iter().for_each(|post| {
+        if let Some(tags) = extract_tags(post) {
+            tags.iter().for_each(|tag| {
+                add_to_tag(
+                    &tag.as_scalar().expect("Should have str tags").to_str(),
+                    post,
+                    &mut per_tags,
+                )
+            })
+        } else {
+            add_to_tag("_none", post, &mut per_tags)
```

Comment:
I'd prefer we use a blank string as a sentinel.  We should also have a constant defined for it.

Reply:
I use this `_none` value to create a directory `tags/_none` if a post has no tag defined in its frontmatter. With an empty string, where will I put such posts?

Reply:
I'll have to think more on this.  I dislike us coming up with our own thing like this.

Reply:
Do you want to discuss this point here or on the RFC there https://github.com/cobalt-org/cobalt.rs/issues/549#issuecomment-438027251 ?

Reply:
Yeah, we should handle it in https://github.com/cobalt-org/cobalt.rs/issues/550 ... once I have a chance to look over the pagination part.

Reply:
I keep this conversion open for when we found a solution and it's implemented :)